packages/hyperion-autologging/src/ALSurfaceData.ts

```diff
@@ -152,6 +155,28 @@ export class ALSurfaceData extends ALSurfaceDataCore {
     }
     surfacesData.set(nonInteractiveSurface, this);
 
+    /**
```

Comment:
I don't know the code well so ignore this if it makes no sense.  Is this not just recreating event bubbling and event capture without handling the corner cases?  If it is, it might be a lot easier and performant to virtualize the events layer than trying to create a parallel one.

Reply:
No, this is not emitting any events. 
We have a `surface` structure in the app, the class in this module is for the nodes of that tree. It is essentially a filtered version of the dom. 

Surface is the only manual thing developers can add to further customize AutoLogging events. Customization happens via metadata. 

The change here allows one to specify what custom metadata should be added to the events that happen under this surface. 

Since surface is hierarchical, a metadata from an ancestor surface should be applied to all events under that surface as well. That's why the logic has this rather complex (and I might say myself: clever!) setup! :)